pkg/fleetautoscalers/controller.go

```diff
@@ -313,7 +313,8 @@ func (c *Controller) syncFleetAutoscaler(ctx context.Context, key string) error
 	}
 
 	currentReplicas := fleet.Status.Replicas
-	desiredReplicas, scalingLimited, err := computeDesiredFleetSize(fas.Spec.Policy, fleet, c.gameServerLister, c.counter.Counts())
+	gameServerNamespacedLister := c.gameServerLister.GameServers(fleet.Namespace)
```

Comment:
Set the lister namespaced at this level as the whole controller is based per namespaces